scripts/website/expand_templates/main.go

```diff
@@ -92,29 +93,35 @@ type latestRelease struct {
 }
 
 func getLatestVersion() (string, error) {
-	req, err := http.NewRequest( //nolint:noctx
-		http.MethodGet,
-		"https://api.github.com/repos/golangci/golangci-lint/releases/latest",
-		http.NoBody,
-	)
+	endpoint := "https://api.github.com/repos/golangci/golangci-lint/releases/latest"
+
+	//nolint:noctx
+	req, err := http.NewRequest(http.MethodGet, endpoint, http.NoBody)
 	if err != nil {
-		return "", fmt.Errorf("failed to prepare a http request: %w", err)
+		return "", fmt.Errorf("prepare a HTTP request: %w", err)
 	}
+
 	req.Header.Add("Accept", "application/vnd.github.v3+json")
 	resp, err := http.DefaultClient.Do(req)
+
 	if err != nil {
-		return "", fmt.Errorf("failed to get http response for the latest tag: %w", err)
+		return "", fmt.Errorf("get HTTP response for the latest tag: %w", err)
 	}
+
 	defer resp.Body.Close()
+
 	body, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return "", fmt.Errorf("failed to read a body for the latest tag: %w", err)
+		return "", fmt.Errorf("read a body for the latest tag: %w", err)
 	}
+
 	release := latestRelease{}
+
 	err = json.Unmarshal(body, &release)
 	if err != nil {
-		return "", fmt.Errorf("failed to unmarshal the body for the latest tag: %w", err)
+		return "", fmt.Errorf("unmarshal the body for the latest tag: %w", err)
 	}
```

Comment:
We can refactor to `json.Decoder`, which will make it four lines shorter (3 for handling `io.ReadAll`, and 1 for importing `io`):
```go
	release := latestRelease{}
	err = json.NewDecoder(resp.Body).Decode(&release)
	if err != nil {
		return "", fmt.Errorf("failed to decode the body for the latest tag: %w", err)
	}
```

Reply:
One of the differences between the `io.ReadAll` + `json.Unmarshal` and `json.NewDecoder().Decode` is the quality of the error.
The errors of `json.NewDecoder().Decode`, if the problem is related to the read of the body, are less relevant than `io.ReadAll`.